Azure/communication-ui-library
# BUG 2602657 : Compliance banner: Added unit test and simplified logic

# What
Moved computeVariant function to utils
Added unit test for computeVariant
Simplified logic for computeVariant, using 3 states: on, off and stopped to determine returned string
<img width="694" alt="Screen Shot 2022-01-17 at 8 57 22 PM" src="https://user-images.githubusercontent.com/96077406/149857500-143987e3-d379-4b4c-b4e8-5a94b1279855.png">
Stopped state means:  Previous state is on, Current state is off 
On state means: Current state is on, regardless of what previous state is
Off state means: previous state is off and current state is still off. Off state is the default state 


# Why
The mobile team is current using the on/off/stop logic, it is more simplied than our logic (using previous and current values), also it is easier for development when mobile and desktop are using the same logic 
Mobile logic for reference: https://github.com/Azure/communication-ui-library-ios/blob/main/AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/Banner/BannerViewModel.swift

# How to test
Start a teams meeting, join teams meeting using Calling sample. start and stop recording / transcribing in teams. Check if the calling sample is displaying correct messages.
